Project-Descriptions-and-Plans/CORE-V-TRUSTED-MCU/Preliminary-project-proposal-for-Trusted-MCU-SoC.md

```diff
@@ -0,0 +1,249 @@
+2. Incorporate a security engine with root of trust and cryptographic services that enables platform level security, for example meeting the requirements of [Matter]https://csa-iot.org/wp-content/uploads/2022/03/Matter_Security_and_Privacy_WP_March-2022.pdf).
+4. The Trusted-MCU SoC targets Functional Safety applications ASIL B compliance. Hence, it supports all the reliability and safety features on the CV32E40S, namely security alert, data independent timing, hardened PC and CSRs, error correction code on register file, interface integrity and bus protocol hardening.
+5. The Trusted-MCU SoC is not meant to be a product but rather a proof of concept. It is intended to be an evaluation device of the CV32E40S, the CV32E20 and the associated peripherals. It is also intended to showcase how commercial SoCs can be built based on the OpenHW Group core-v-mcu platform.
+
+## Components
+The project comprises two main components.
+
+### Component 1 - Design and Fabrication of the SoC Device
+* Design [effort | timeline]
+    - Interconnect update to include address filtering and integrity check
+    - Power Management: Define and Partition the SoC into power domains
+    - Enable Priority based arbitration mechanism on the TCM memory interface
+* Integration [effort | timeline]
+    - Repository for the Trusted MCU RTL
+    - Event unit update
+    - FPGA Implementation
+        - Synthesis and bitstream generation
+        - Test firmware
+    - ASIC Implementation
+        - Synthesis
+        - Power Management - UPF for Power Gating
+        - RTL to GDS2 conversion
+* UVM Verification [effort | timeline]
+    - uDMA verification and associated peripherals
+        - Verification Plan
+        - Register Model
+        - UVM test benches and environment
+    - APB peripheral verification (Timer, Advanced Timer, I2C, UART, etc.)
+        - Verification Plan
+        - Register Model
+        - UVM test benches and environment
+    - SoC top level verification
+        - Replace the two cores with OBI agents
+        - Test hardware reset and register bash for all IPs
+* Firmware [effort | timeline]
+    - Secure boot firmware 
+    - Firmware signature verification using both symmetric and asymetric cryptographic algorithms
+    - Interprocessor communication
+
+### Component 2 - Design and Build of the Evaluation Kit
+[Detailed timeline to be provided]
+* Board based on the MCU
+* SDK based on the SDK of the MCU
+* Documentation: lightweight user guide of the device such that a competent hardware or software engineer can evaluate the device on the dev kit.
+
+### Summary of Timeline
+
+* Architecture			Q4 2022
+* Specification			Q1 2023
+* RTL Integration
+* FPGA Prototyping		Q1 2023
+* UVM Verification
+* RTL to GDS			Q2 2023
+* Development kit
+* Firmware
+* Tapeout				Q3 2023
```

Comment:
Based on the MCU timeline, this seems wildly optimistic

Reply:
I agree, but let's keep it for now.  Once we see the task breakdown we'll have a better idea.